benches/transform/dedupe.rs

```diff
@@ -0,0 +1,123 @@
+use crate::common::{consume, FixedLogStream};
+use core::fmt;
+use criterion::BenchmarkId;
+use criterion::{
+    criterion_group, measurement::WallTime, BatchSize, BenchmarkGroup, Criterion, SamplingMode,
+    Throughput,
+};
+use std::num::NonZeroUsize;
+use std::time::Duration;
+use vector::transforms::dedupe::{CacheConfig, Dedupe, DedupeConfig, FieldMatchConfig};
+use vector_core::transform::Transform;
+
+#[derive(Debug)]
+struct Param {
+    slug: &'static str,
+    input: FixedLogStream,
+    dedupe_config: DedupeConfig,
+}
+
+impl fmt::Display for Param {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.slug)
+    }
+}
+
+fn dedupe(c: &mut Criterion) {
+    let mut group: BenchmarkGroup<WallTime> =
+        c.benchmark_group("vector::transforms::dedupe::Dedupe");
+    group.sampling_mode(SamplingMode::Auto);
+
+    let fixed_stream = FixedLogStream::new(
+        NonZeroUsize::new(128).unwrap(),
+        NonZeroUsize::new(2).unwrap(),
+    );
+    for param in &[
+        // Measurement where field "message" is ignored. This field is
+        // automatically added by the LogEvent construction mechanism.
+        Param {
+            slug: "field_ignore_message",
+            input: fixed_stream.clone(),
+            dedupe_config: DedupeConfig {
+                fields: Some(FieldMatchConfig::IgnoreFields(vec![String::from(
+                    "message",
+                )])),
+                cache: CacheConfig { num_events: 4 },
```

Comment:
It might be more in the spirit of this PR to use the default. I'd intended to drive aberrant behavior with smaller cache size, but the PR is generally focused on having something we can eventually mate up with soak tests.